src/main/webapp/app/overview/course-conversations/dialogs/forward-message-dialog/forward-message-dialog.component.ts

```diff
@@ -0,0 +1,238 @@
+import { AfterViewInit, ChangeDetectorRef, Component, ElementRef, HostListener, OnInit, Renderer2, inject, input, signal, viewChild } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ChannelDTO } from 'app/entities/metis/conversation/channel.model';
+import { Post } from 'app/entities/metis/post.model';
+import { BoldAction } from 'app/shared/monaco-editor/model/actions/bold.action';
+import { ItalicAction } from 'app/shared/monaco-editor/model/actions/italic.action';
+import { UnderlineAction } from 'app/shared/monaco-editor/model/actions/underline.action';
+import { QuoteAction } from 'app/shared/monaco-editor/model/actions/quote.action';
+import { CodeAction } from 'app/shared/monaco-editor/model/actions/code.action';
+import { CodeBlockAction } from 'app/shared/monaco-editor/model/actions/code-block.action';
+import { UrlAction } from 'app/shared/monaco-editor/model/actions/url.action';
+import { TextEditorAction } from 'app/shared/monaco-editor/model/actions/text-editor-action.model';
+import { MarkdownEditorHeight } from 'app/shared/markdown-editor/monaco/markdown-editor-monaco.component';
+import { UserPublicInfoDTO } from 'app/core/user/user.model';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { catchError, map, of } from 'rxjs';
+
+interface CombinedOption {
+    id: number;
+    name: string;
+    type: string;
+    img: string;
+}
+
+@Component({
+    selector: 'jhi-forward-message-dialog',
+    templateUrl: './forward-message-dialog.component.html',
+    styleUrls: ['./forward-message-dialog.component.scss'],
+})
+export class ForwardMessageDialogComponent implements OnInit, AfterViewInit {
+    channels = signal<ChannelDTO[] | []>([]);
+    users = signal<UserPublicInfoDTO[] | []>([]);
+    postToForward = signal<Post | null>(null);
```

Comment:
we prefer `undefined` over `null` in the client code, see the client coding guidelines